src/breadcrumb.rs

```diff
@@ -28,8 +28,12 @@ impl Breadcrumb {
     /// Creates a new Sentry breadcrumb.
     #[must_use]
     pub fn new(r#type: Option<SentryString>, message: Option<SentryString>) -> Self {
```

Comment:
```suggestion
    pub fn new(r#type: Option<&SentryString>, message: Option<&SentryString>) -> Self {
```

Because they aren't being consumed anyway, this would remove the need for `as_ref`, yes?

Reply:
That would also work though it makes it a bit more awkward to use as these strings are usually constructed at the callsite and used once.

Reply:
You are right, I will look into this more after I remove the whole 0 byte handling and try to improve the API usage.